### ISSUE TITLE
Gmail Labels not Updating in Nylas N1

### ISSUE THREAD

**Issue by merriweather707** [nylas/N1#2451]
Changes to labels in Gmail (or in Airmail, syncing back to Gmail) are not properly reflected in Nylas. The behavior I am seeing is as follows:

If I make a new label in Gmail, it is synced to N1 and appears as expected.
If I make a new label in N1, it is synced to Gmail and appears as expected.

If I edit the name of an existing label in Gmail, the change does not sync to N1. 
If I edit the name of an existing label in N1, the change syncs to Gmail as expected.

If I delete an existing label in Gmail, the label is not deleted in N1.
If I delete an existing label in N1, the label is deleted in Gmail.

If I change a label in Gmail so that it is nested, or change the folder it is nested under, that change is not reflected in N1. 

Reloading N1, restarting N1, and deleting the cache from the Application Support folder have no effect.

If I completely remove the account from N1 and add it back, the labels properly populate on the first sync.

All my Gmail folders are set to appear in IMAP. There is no message limit set for IMAP sync.

I am running OSX 10.11.4
N1 version 0.4.45-7637265 (0.4.45-7637265)

![label comparison](https://cloud.githubusercontent.com/assets/19955198/16086296/0a07f582-32ed-11e6-8bcf-05857a8ca7be.png)

**Comment by Tsmith18256** [comment]
+1 on OS X and Windows. I think there was an issue logged for this a long time ago that I also commented on, but can't remember. I have been having the issue for a while.

**Comment by vallettea** [comment]
same problem

**Comment by vallettea** [comment]
Is somebody looking into this ? It's really a big problem since other mailers (on mobile phones for example) display all the mails that where not correctly archived or trashed by nylas.

**Comment by Tsmith18256** [comment]
Yeah this has made the program unusable to me. I was using N1 for a while and loved it, but haven't been able to use it since around May because I have tons of emails I need access too in labels that aren't showing up in N1. It's sad to see such a great email app that I saw so much potential in be completely hindered by such a large bug that doesn't seem to be very high priority.

**Comment by hallamoore** [comment]
Some, if not all, of these scenarios should be fixed. Are there any scenarios that are still having issues?

**Comment by vallettea** [comment]
Thanks for your work, however I encounter the same problem with the last version of N1.

**Comment by Tsmith18256** [comment]
I just added my personal Gmail account to the instance of N1 running on my work computer with the newest update that downloaded for me this morning. The account has never been on this computer so caching isn't an issue, and it only brought down 2 of my labels (not counting Sent, Drafts and Trash). There are many others I have that should be included, too.

**Comment by jackiehluo** [comment]
Another incidence: https://nylas.zendesk.com/agent/tickets/5430

**Comment by richarddong** [comment]
+1
